openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -515,3 +573,110 @@ def partial_update(self, request):
             return Response(text_type(exc), status=status.HTTP_400_BAD_REQUEST)
         except Exception as exc:  # pylint: disable=broad-except
             return Response(text_type(exc), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    @request_requires_username
+    def post(self, request):
+        """
+        POST /api/user/v1/accounts/retire/
+
+        {
+            'username': 'user_to_retire'
+        }
+
+        Retires the user with the given username.  This includes
+        retiring this username, the associates email address, and
+        any other PII associated with this user.
+
+        Note that this implementation is the "merge patch" implementation proposed in
+        https://tools.ietf.org/html/rfc7396. The content_type must be "application/merge-patch+json" or
+        else an error response with status code 415 will be returned.
+        """
+        username = request.data['username']
+        if is_username_retired(username):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        user_model = get_user_model()
```

Comment:
Since this is retirement-specific I'd recommend getting at the user via the retirement, which will do some basic status checking and confirm that this is a user who is actually in the retirement process. `UserRetirementStatus::get_retirement_for_retirement_action(username)` should get you their `UserRetirementStatus` record, if it exists, and you can get the user from there.